### PR TITLE
Adds an EngineBuilder API to set restart features

### PR DESCRIPTION
**Pull request by abeyad**

### CODE REVIEW

File: mobile/library/cc/engine_builder.h [hunk 1]
```diff
@@ -89,6 +89,11 @@ class EngineBuilder {
   // For example if the runtime guard is `envoy.reloadable_features.use_foo`, the guard name is
   // `use_foo`.
   EngineBuilder& addRuntimeGuard(std::string guard, bool value);
```
Comment (fredyw) [inline mobile/library/cc/engine_builder.h:91]: I wish we could rename this to `addRelodableRuntimeGuard` :)

**Review by fredyw — APPROVED** [review]

File: mobile/library/cc/engine_builder.h (hunk 1 above)
Comment (abeyad) [reply mobile/library/cc/engine_builder.h:91]: yeah, in a follow up :)  i wanted to do the same but it'll break existing code.  i'll probably add addReloadableRuntimeGuard, then when the code is switched over to using it, delete addRuntimeGuard

**Pull request merged**